### ISSUE TITLE
Search Results - Add pagination

### ISSUE THREAD

**Issue by samjiangweb** [participedia/frontend#192]
Let's move away from the "click to view more" approach and add in pagination to the search results.

Let's show a default view of say.. 20, and provide a dropdown that lets users select how many results per page (placed at the bottom of results - next to pagination) - Content could be: "Show [20 dropdown] results per page"

**Comment by davidascher** [comment]
Flagging here that we discussed in IRC the question of what it means to paginate on a mixed-type view, and that @samjianga will come back with a specific proposal.

**Comment by dethe** [comment]
The search API supports this now, pass `&page=[positive integer]`, with two caveats:

1. Current page size is 30, not 20 (easily changed)
2. If object type is `All` (vs. `Case`, `Method`, etc.) then 30 (or PAGE_SIZE) of *each* object type is returned. Returning an interleaved set, interleaved by selected sort order, in combination with paging is a non-trivial problem.

**Comment by samjiangweb** [comment]
Let's avoid non-trivial work for the "All" filter - this view actually has very little value for users. That being said, if we can mix results, instead of section by "Case" "Method" etc... I understand that most items will be cases by default, but the only real value we'd get from this view is perhaps the "featured" items which we're still deciding on.

**Comment by davidascher** [comment]
OK.  @dethe, let's change the semantics of the current search API `selectedCategory=All` to mean interleave.

**Comment by jesicarson** [comment]
@samjianga ignore me if you are already on this, but there was mention in the meeting today about the need for design related to options for selecting previous / next / first page / last page on search results. Thanks!

**Comment by davidascher** [comment]
Needs some styling decisions, but also need https://github.com/participedia/api/issues/84

**Comment by samjiangweb** [comment]
![screen shot 2017-05-09 at 3 57 58 pm](https://cloud.githubusercontent.com/assets/25598431/25876170/506435b4-34d0-11e7-8796-5cf1f780f9e7.png)

@davidascher @andreadelrio @jesicarson - this is what I have in terms of structure. The design can be dealt with in browser. Might be better to think of this screen as a "wireframe" for the content of the pagination piece.

**Comment by jesicarson** [comment]
@samjianga what do you mean by "the design can be dealt with in browser"? Thanks!

**Comment by samjiangweb** [comment]
@jesicarson - If we're not cool with that design - I can edit the CSS directly to change the visual designs, rather than making a series of mockups.

**Comment by jesicarson** [comment]
@samjiangweb Gotcha, thanks!

**Comment by jesicarson** [comment]
No pagination on xyz as of today

**Comment by jesicarson** [comment]
![screen shot 2017-10-04 at 2 26 04 pm](https://user-images.githubusercontent.com/13560530/31200499-03b53f7e-a910-11e7-80d2-471549a5c617.png)

This works well, but we should show the number of results (ie: 1-20) out of the total number of content items in each category (all, cases, methods, orgs) instead of the number of pages. so for cases it would be "1-20 of 735" . Also, is 20 too few results per page? I think maybe 40 would be better.

**Comment by davidascher** [comment]
The current site shows

![image](https://user-images.githubusercontent.com/51278/32703830-4e2a702e-c7b0-11e7-8a4c-4e0621c019fa.png)

is that ok?

Changing page size from 20 to 40 is easy, let me know if you still want that @jesicarson.

**Comment by jesicarson** [comment]
@davidascher yes that works perfectly for results on a search term. However, on the homepage, if I haven't searched for anything, it just shows "page x of x", not the number of results. I want to be able to see the total number of articles on the site (on the "all" tab), and individually the total number of cases, methods and orgs on those tabs as well. 
So on home it should be "xx Results (page x of x)" 
And yes, 40 results per page would be good please. 
Thanks!

**Comment by dethe** [comment]
If that's a requirement, add an issue for it. The back-end already supports passing an option for how many results to return. Not infinite scroll though, that is an abomination that should have been drowned at birth.

I think anything besides picking a reasonable default is probably overkill, but maybe that's just me.

**Comment by scottofletcher** [comment]
Ahaha! I always choose infinite scroll (and will continue to do so just to spite that comment!)

Sure, I'll add an issue for this but it's totally P2 (or Pnever)
